lib/cdo/pardot.rb

```diff
@@ -195,24 +194,9 @@ def self.sync_contacts_with_pardot(config)
           prospect[pardot_info[:field]] = db_value
         end
       end
-      # Special case: if contact has opted out, set the two different Pardot
-      # flavors of opted out. Also, only ever set this to true, otherwise set no
-      # value; never set it back to false. Pardot is the authority on opt-out
-      # data, so never reset any opt-out setting it has stored.
-      if contact_rollup[:opted_out]
-        prospect[:opted_out] = true
-        prospect[:is_do_not_email] = true
-      end
-
-      # If this contact has a dashboard user ID (which means it is a teacher
-      # account), mark that in a Pardot field so we can segment on that.
-      if contact_rollup[:dashboard_user_id].present?
-        prospect[:db_Has_Teacher_Account] = "true"
-      end
+      # Do special handling of a few fields
+      apply_special_fields(contact_rollup, prospect)
```

Comment:
This code moved to a method so it can also be called from `gen-pardot-csv` script.